Sources/SwiftCompilerPlugin/Macros.swift

```diff
@@ -0,0 +1,254 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+/// Implementation for `CompilerPlugin` macro related request processing.
+extension CompilerPlugin {
+  public  // @testable
+    func resolveMacro(moduleName: String, typeName: String) -> Macro.Type?
```

Comment:
What I’ve done before (and what I’m relatively happy with) to make methods public for testing is to define the real function as internal and then add a public thunk around it that is prefixed with `_` and marked as only being public for testing purposes.